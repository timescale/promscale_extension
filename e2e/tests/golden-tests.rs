use crate::common::{exec_sql_script, run_postgres};
use insta::assert_snapshot;
use std::env;
use test_generator::test_resources;
use testcontainers::clients;

mod common;

#[test_resources("testdata/*.sql")]
fn golden_test(resource: &str) {
    let docker = clients::Cli::default();
    let node = run_postgres(&docker);
    let query_result = exec_sql_script(&node, resource);

    assert_snapshot!(resource, query_result);
}
