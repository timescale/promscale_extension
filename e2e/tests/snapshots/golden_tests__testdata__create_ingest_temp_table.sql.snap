---
source: e2e/tests/golden-tests.rs
expression: query_result
---
                     get_or_create_metric_table_name                      
--------------------------------------------------------------------------
 (1,cpu_usage,t)
 (2,"aMultiCase""fun""Metric!",t)
 (3,"aVeryLongMultiCaseMetricWhich""might""BeTruncatedIfIt'sTOOOOOO_3",t)
(3 rows)

                                               runtests                                               
------------------------------------------------------------------------------------------------------
     # Subtest: public.test_create_ingest_temp_table()
     ok 1 - table prom_data.cpu_usage exists
     ok 2 - temp table name is well-formed
     ok 3 - temp table exe_cpu_usage exists
     ok 4 - table prom_data."aMultiCase""fun""Metric!" exists
     ok 5 - temp table name is well-formed
     ok 6 - temp table "eXe_aMultiCase""fun""Metric!" exists
     ok 7 - table prom_data."aVeryLongMultiCaseMetricWhich""might""BeTruncatedIfIt'sTOOOOOO_3" exists
     ok 8 - temp table name is well-formed
     ok 9 - temp table "ThisIsLong_aVeryLongMultiCaseMetricWhich""might""BeTruncatedIfIt" exists
     1..9
 ok 1 - public.test_create_ingest_temp_table
 1..1
(13 rows)


